sparrow-py/pysrc/sparrow_py/_ffi.pyi

```diff
@@ -8,6 +8,9 @@ from sparrow_py.udf import Udf
 class Session:
     def __init__(self) -> None: ...
 
+class ExecutionOptions(object):
+    def __init__(self, row_limit: Optional[int] = None) -> None: ...
+    row_limit: Optional[int]
```

Comment:
I don't know python, but chatgpt said this looks a lot like a data class and can be done (in 3.7 >)
```
from dataclasses import dataclass

@dataclass
class ExecutionOptions:
    row_limit: Optional[int] = None
```

Reply:
Yeah. But this is currently the type signature (`pyi` not `py`) for a class that is defined in Rust code. We could maybe go the other way, but then we'd need to do a lot more casting / conversion that pyo3 defines for us this way.

Reply:
I guess we could do that though -- we could do:

1. Define the dataclass only in python
2. Pass that to rust.
3. Use https://docs.rs/pyo3/latest/pyo3/types/struct.PyAny.html#method.getattr to convert:

```
let row_limit: Option<usize> = options.getattr("row_limit").extract()?;
...
```

As we have more options that would decrease duplication. Leave open and I'll add this to the PR.

Reply:
ohh got it okay